app/javascript/components/upload/MetadataStep.js

```diff
@@ -0,0 +1,141 @@
+import React, { useEffect, useContext } from 'react'
+import metadataExplainerImage from 'images/metadata-convention-explainer.jpg'
+import { faInfoCircle } from '@fortawesome/free-solid-svg-icons'
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import { Popover, OverlayTrigger } from 'react-bootstrap'
+
+import { UserContext } from 'providers/UserProvider'
+import FileUploadControl, { FileTypeExtensions } from './FileUploadControl'
+import { TextFormField, SavingOverlay, SaveDeleteButtons } from './uploadUtils'
+
+const DEFAULT_NEW_METADATA_FILE = {
+  file_type: 'Metadata',
+  use_metadata_convention: true
+}
+
+const metadataFileFilter = file => file.file_type === 'Metadata'
+
+export default {
+  title: 'Metadata',
+  name: 'metadata',
+  component: MetadataForm,
+  fileFilter: metadataFileFilter
+}
+
+/** Renders a form for uploading one or more cluster/spatial files */
+function MetadataForm({
+  formState,
+  addNewFile,
+  updateFile,
+  saveFile,
```

Comment:
This works as advertised, but I feel like there is a UX regression here in terms of a successful upload/save.  There is no visual indication to the user that the action succeeded.  I'm not necessarily advocating for a modal (we have far too many, and they are very obtrusive), but the current UX (non-React) is _very_ explicit when things work, and when they don't.  I'd like to find a middle ground that gives necessary feedback to the user without taking over the page.

Reply:
yup, reasonable success/fail/parsing alerts haven't been done yet -- Jerome is still brainstorming concepts.  I'll create a separate ticket to make sure it gets addressed